### PR TITLE
Fix/adf 12/code integration

### PR DESCRIPTION
**Pull request by boajer**
https://oat-sa.atlassian.net/browse/ADF-12

connected to:

- [ ] https://github.com/oat-sa/tao-core/pull/2835
- [ ] https://github.com/oat-sa/extension-tao-dac-simple/pull/151
- [ ] https://github.com/oat-sa/extension-tao-outcomeui/pull/376
- [ ] https://github.com/oat-sa/extension-tao-advanced-search/pull/12
- [ ] https://github.com/oat-sa/lib-tao-elasticsearch/pull/44
- [ ] https://github.com/oat-sa/extension-tao-revision/pull/122 needs to be merged to develop too
- [ ] https://github.com/oat-sa/extension-tao-publishing/pull/104 needs to be merged to develop too

todo:

- [ ] ensure proper versioning?

### CODE REVIEW

File: config/default/eventAggregator.conf.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ * @license GPLv2
+ * @package generis
+ *
+ */
+
+use oat\oatbox\event\EventAggregator;
+
+return new EventAggregator(
+    [
+        'numberOfAggregatedEvents' => 10,
```
Comment (shpran) [inline config/default/eventAggregator.conf.php:30]: Are you sure that the number of aggregated events should be 10 ❔

File: core/data/permission/ReverseRightLookupInterface.php [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+declare(strict_types=1);
+
+namespace oat\generis\model\data\permission;
+
+/**
+ * ReverseRightLookupInterface
+ *
+ * @package oat\generis\model\data\permission
+ */
```
Comment (shpran) [inline core/data/permission/ReverseRightLookupInterface.php:29]: I don't think we need this information 🙂

File: core/data/permission/ReverseRightLookupInterface.php [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+     *        'WRITE'
+     *    ],
+     *    'http://www.tao.lu/Ontologies/TAO.rdf#ItemAuthor' => [
+     *        'GRANT',
+     *        'READ'
+     *    ],
+     * ]
+     *
+     * @param string $resourceId
+     * @return array
```
Comment (shpran) [inline core/data/permission/ReverseRightLookupInterface.php:50]: Can be removed, we can find this information in method definition

File: migrations/Version202008201138552348_generis.php [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        return 'New EventAggregator Service for extension generis.';
+    }
+
+    public function up(Schema $schema): void
+    {
+        $eventAggregator = new EventAggregator(
+          [
+              'numberOfAggregatedEvents' => 10
+          ]
+        );
```
Comment (shpran) [inline migrations/Version202008201138552348_generis.php:27]:
```suggestion
        $eventAggregator = new EventAggregator([
            'numberOfAggregatedEvents' => 10,
        ]);
```

File: test/OntologyMockTrait.php [hunk 5]
```diff
@@ -74,29 +75,33 @@ protected function getOntologyMock()
      */
     private function setupOntology(Ontology $onto)
     {
-        $pm = $this->getPersistenceManagerWithSqlMock('mockSql');
+        $eventAggregator = new EventAggregator(['numberOfAggregatedEvents'=>10]);
```
Comment (shpran) [inline test/OntologyMockTrait.php:78]:
```suggestion
        $eventAggregator = new EventAggregator(['numberOfAggregatedEvents' => 10]);
```

**Review by shpran — APPROVED** [review]
Left several CS comments 🙂 
Remember, that we can remove doc-blocs (fully or partially) if the method definition contains the necessary info.

- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Review by voknil — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

File: config/default/eventAggregator.conf.php (hunk 1 above)
Comment (boajer) [reply config/default/eventAggregator.conf.php:30]: Nope, but that is configurable so can be changed without redeployment if needed  or benchmarking will show a need

**Review by GregoryGagua — APPROVED** [review]

**Pull request merged**